spec.bs

```diff
@@ -243,8 +250,200 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
+  1. If |init|'s {{URLPatternInit/baseURL}} is not null:
+    1. Let |baseURL| be the result of [=URL parser|parsing=] |init|'s {{URLPatternInit/baseURL}}.
+    1. If |baseURL| is failure, then throw a TypeError.
+    1. Set |result|'s {{URLPatternInit/protocol}} to |baseURL|'s [=URL/protocol].
+    1. Set |result|'s {{URLPatternInit/username}} to |baseURL|'s [=URL/username].
+    1. Set |result|'s {{URLPatternInit/password}} to |baseURL|'s [=URL/password].
+    1. Set |result|'s {{URLPatternInit/hostname}} to |baseURL|'s [=URL/hostname].
+    1. Set |result|'s {{URLPatternInit/port}} to |baseURL|'s [=URL/port].
+    1. Set |result|'s {{URLPatternInit/pathname}} to |baseURL|'s [=URL/pathname].
+    1. Set |result|'s {{URLPatternInit/search}} to |baseURL|'s [=URL/search].
+    1. Set |result|'s {{URLPatternInit/hash}} to |baseURL|'s [=URL/hash].
+  1. If |init|'s {{URLPatternInit/protocol}} is not null:
+    1. Set |result|'s {{URLPatternInit/protocol}} to the result of [=canonicalize protocol=] given |init|'s {{URLPatternInit/protocol}} and |type|.
+  1. If |init|'s {{URLPatternInit/username}} is not null:
+    1. Set |result|'s {{URLPatternInit/username}} to the result of [=canonicalize username=] given |init|'s {{URLPatternInit/username}} and |type|.
+  1. If |init|'s {{URLPatternInit/password}} is not null:
+    1. Set |result|'s {{URLPatternInit/password}} to the result of [=canonicalize password=] given |init|'s {{URLPatternInit/password}} and |type|.
+  1. If |init|'s {{URLPatternInit/hostname}} is not null:
+    1. Set |result|'s {{URLPatternInit/hostname}} to the result of [=canonicalize hostname=] given |init|'s {{URLPatternInit/hostname}} and |type|.
+  1. If |init|'s {{URLPatternInit/port}} is not null:
+    1. Set |result|'s {{URLPatternInit/port}} to the result of [=canonicalize port=] given |init|'s {{URLPatternInit/port}} and |type|.
+  1. If |init|'s {{URLPatternInit/pathname}} is not null:
+    1. Set |result|'s {{URLPatternInit/pathname}} to the result of [=canonicalize pathname=] given |init|'s {{URLPatternInit/pathname}} and |type|.
+  1. If |init|'s {{URLPatternInit/search}} is not null:
+    1. Set |result|'s {{URLPatternInit/search}} to the result of [=canonicalize search=] given |init|'s {{URLPatternInit/search}} and |type|.
+  1. If |init|'s {{URLPatternInit/hash}} is not null:
+    1. Set |result|'s {{URLPatternInit/hash}} to the result of [=canonicalize hash=] given |init|'s {{URLPatternInit/hash}} and |type|.
+  1. Return |result|.
+</div>
+
+<div algorithm>
+  To <dfn>create a component match result</dfn> given a [=component=] |component|, a string |input|, and an array representing the output of [=RegExpExec=] |execResult|:
+
+  1. Let |result| be the result of creating a new {{URLPatternComponentResult}}.
+  1. Set |result|'s {{URLPatternComponentResult/input}} to |input|.
+  1. Let |groups| be an empty object.
+  1. Let |index| = 1;
+  1. While |index| is less than |execResult|.length:
```

Comment:
You can't do property access directly here. You need to use `[$Get$](|execResult|, "`length`")` with the following in your anchors block:

```
urlPrefix: https://tc39.es/ecma262/; spec: ECMASCRIPT
 type: abstract-op
  text: Get; url: #sec-get-o-p
```

Same below for the indexed properties. (Although there you'll also need to do `[$ToString$](|index|)` which will need a similar anchor.